### DISCUSSION TITLE
Why aren't migrations split into up.sql and down.sql?

### DISCUSSION THREAD

**Discussion by hariria** [prisma/prisma#20957] [Q&A]
### Question

Hi, just wanted to ask why database migrations in prisma don't use the `up.sql` + `down.sql` pattern and instead opt for one `migration.sql`? And might there be a way to customize it in the future? I wish there was a little more granularity for more complex migrations. Other ORMs like diesel.rs use this pattern as well.

**Comment by nurul3101** [comment]
Hello @hariria 👋 

You can generate down migrations ([Reference](https://www.prisma.io/docs/guides/migrate/developing-with-prisma-migrate/generating-down-migrations)) for every generated `migration.sql` file. It is not created by default. 

> And might there be a way to customize it in the future? I wish there was a little more granularity for more complex migrations.

Could you please elaborate on what kind of granularity you would like?

**Comment by hariria** [comment]
> You can generate down migrations ([Reference](https://www.prisma.io/docs/guides/migrate/developing-with-prisma-migrate/generating-down-migrations)) for every generated migration.sql file. It is not created by default.

I see got it thanks

> Could you please elaborate on what kind of granularity you would like?

I wish users could choose between the default `migration.sql` and `up.sql` / `down.sql` pattern and decide which one they'd like to use as a default rather than doing this `npx prisma migrate diff` command that produces a down.sql in a different folder and then requires you to manually make changes. 

The default would be nice because if I have a complex schema design with DB triggers and indexes, I want to have clear, fine-grained control over my db changes that are committed to version control and be able to rollback changes easily if need be.

As a bonus, the `up.sql` and `down.sql` pattern would also make migrations from other ORMs easier. Almost every other production ORM I've seen uses it. As an example, Typeorm and diesel both use the `up.sql`, `down.sql` pattern.

Curious if this could be done. It doesn't seem too difficult to implement and I think would be a huge value add to the community Cc @ruheni @tomhoule

**Reply by hariria** [reply]
Also seems like I'm not the only one who feels this way (https://github.com/prisma/prisma/discussions/4617#discussioncomment-6458346)

**Comment by hariria** [comment]
One more thing @nurul3101 , it doesn't seem like you can generate a `down.sql` for the first migration? Only subsequent ones?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
